modules/plugins/utility/harpoon/config.nix

```diff
@@ -0,0 +1,51 @@
+{
+  options,
+  config,
+  lib,
+  ...
+}: let
+  inherit (lib.modules) mkIf;
+  inherit (lib.lists) optionals;
+  inherit (lib.strings) concatStrings;
+  inherit (lib.nvim.binds) pushDownDefault mkKeymap;
+
+  cfg = config.vim.navigation.harpoon;
+
+  keys = cfg.mappings;
+  inherit (options.vim.navigation.harpoon) mappings;
+in {
+  config = mkIf cfg.enable {
+    vim = {
+      startPlugins = ["plenary-nvim"];
+
+      lazy.plugins.harpoon = {
+        package = "harpoon";
+        setupModule = "harpoon";
+        inherit (cfg) setupOpts;
+
+        before = ''
+        '';
+
+        after = ''
+          local harpoon = require("harpoon")
+          harpoon.setup({})
+        '';
```

Comment:
The `inherit (cfg) setupOpts;` line already handles this, you may safely omit this since the setup function will be constructed from `vim.navigation.harpoon` and added to the Lua configuration.